packages/chat-core/src/index.ts

```diff
@@ -303,6 +317,15 @@ export const createConversation = (config: Config): ConversationHandler => {
     null;
 
   const sendToBot = async (body: BotRequest): Promise<void> => {
+    const lastExpirationTimestamp = getLastExpirationTimestamp(state.responses);
+    if (
+      typeof lastExpirationTimestamp === "number" &&
+      new Date().getTime() > lastExpirationTimestamp
+    ) {
+      setState({
```

Comment:
💭 this seems like it should raise an exception or signal in some way that there's a new conversation starting

Reply:
@michaelglass totally with you there, we are discussing the behind-the-scenes way to do this, basically any 'your conversation has expired' type of text content needs to be controlled inside the platform, so you can imagine some kind of explicit special message passing will happen here, but later.

Reply:
Can we add a `conversationEnded` subscription thinger here.

Reply:
@sam-trost that sounds like a bit of a change in direction, I would rather have a discussion about it at some point this week.